common/script/cron.js

```diff
@@ -88,6 +88,21 @@ export function daysSince (yesterday, options = {}) {
   Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
  */
 
+function _lastCompletedCheck (dailyHistory) {
+  _.reverse(dailyHistory); // Work in reverse chronological order
+  let nextIndex = 1;
+  for (let entry of dailyHistory) {
+    if (!dailyHistory[nextIndex]) {
+      return false; // History is empty or value static throughout; Daily was never completed or failed
```

Comment:
Thus `nextIndex`, not current index, which is handled within the iterator. We want to quit out if there's nothing to compare our current index against, which is the case when:

a) There is no history data at all
b) There is only one history entry
c) We've run to the end of the history without finding any entries whose values differ

I think the logic I wrote catches those? Though I am very likely wrong, and happy to be shown how, ha.